### PR TITLE
ConversationRepository&Core: all implemented; lighter core models

### LINKED ISSUE #13: ConversationRepository: all features
- [x] Implement the abstract class `ConversationRepository` in the core, the necessary DTOs, and the corresponding SQLA class in infrastructure

Then, implement all planned Conversation features for the ConversationRepository. They are the following:

- [x] NewConversation: `POST`; Start a new conversation, assumes an the llm model was selected
- [x] GetConversationResearchContext: `GET`; Shows the research context of a conversation
- [x] ListConversationMessages: `GET`; List the messages of a conversation
- [x] UpdateConversation: `PUT`; Update the title of a conversation
- [x] ~ListConversations: `GET`; List all conversations in a research context~ *to be implemented in ResearchContextRepository
- [x] ListConversationSources: `GET`; List the source data, used in the research context, that were used to generate a message

For each of these features:

- Implement the corresponding method in `ConversationRepository` core
- Implement the SQLA method in infrastructure
- Test

### LINKED ISSUE #14: core: lighter models
- [x] Take out all of the attributes in the core entities that tie them to other entities in the core. This way we can have lighter methods for repositories and gateways, that don't need to load many objects
- [x] Change the BaseDTO class to contain a data: Optional attribute using Generics to represent any core entity
- [x] Change the SQLA models and the alembic migration file to represent the new changes

### PR DESCRIPTION
**Pull request by alebg**
Fixes #13
Fixes #14

### CODE REVIEW

File: lib/core/dto/.gitkeep
Comment (maany) [inline lib/core/dto/.gitkeep:1]: I guess this can be removed now?

**Review by maany — APPROVED** [review]
Ready to merge!! Awesome work @alebg !!!

File: lib/core/dto/.gitkeep
Comment (alebg) [reply lib/core/dto/.gitkeep:1]: Oops, you're right. Will do for the next PR

Comment (alebg) [reply lib/core/dto/.gitkeep:1]: Actually I did delete it, this is in red because of that :p

**Pull request merged**